src/Lecture3.hs

```diff
@@ -178,6 +206,7 @@ together only different elements.
 Product {getProduct = 6}
 
 -}
+appendDiff3 :: (Eq a, Semigroup a) => a -> a -> a -> a
 appendDiff3 = error "TODO"
 
```

Comment:
Unfortunately, I don't understand how to delete duplicates with type classes, which we know

Reply:
You don't need to delete duplicates 🙂 
Your type signature is absolutely correct! 💯 
What you need is just not to add equal elements. This can be solved with something like guards where you can perform some checks and append only not-equal elements.